### PR TITLE
Restore various default workspaces by using menu

### PR DESCRIPTION
**Pull request by demvlad**
There are two various default workspaces  in the current release and nightly  Blackbox explorers versions at this moment (@SupaflyFPV, @ctzsnooze workspaces) .
Anyway, it is unpossible to restore default workspace after change. 
It is necessary to edit some system files to show default workspace again.
On the other hand, there were suggestion to do possible using of several default workspaces (@mituritsyn).
This PR do that.
Press _Shift+W_ keys to show menu and select workspace preset what you need in any time.
ATTENTION!!!! Your current workspace will be rewritten after that. Save it before!
ws_menu

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1948,6 +1951,11 @@ function BlackboxLogViewer() {
                         }
                         e.preventDefault();
                     break;
+                    case "W".charCodeAt(0):
+                        if(e.shiftKey) {
```
Comment (haslinghuis) [inline src/main.js:1955]: :space_invader: 

```suggestion
                        if (e.shiftKey) {
```

File: src/workspace_menu.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            break;
+            case 2:
+                onSwitchWorkspace(supaflyWorkspace, 1);
+            break;
+        }
+        hideMenu();
+    }
+
+    $(document).keydown( function (e) {
+        if (e.which == 27 && workspace_menu.length>0) {
```
Comment (haslinghuis) [inline src/workspace_menu.js:43]:
```suggestion
        if (e.which === 27 && workspace_menu.length > 0) {
```

File: src/workspace_menu.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            e.preventDefault();
+            hideMenu();
+        }
+    });
+
+    this.getDefaultWorkspace = function() {
+        return ctzsnoozeWorkspace;
+    };
+}
```
Comment (haslinghuis) [inline src/workspace_menu.js:52]: Please add new line on EOL
```suggestion
}

```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: src/workspace_menu.js (hunk 3 above)
Comment (demvlad) [reply src/workspace_menu.js]: > Please add new line on EOL

Ok. Will check.
It seems, i've caught other new line from .json exported file.

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
This PR merged to master has seemed to break loading BBL files.

to reproduce:
option 1) Netlify link above: open file. broken.
option 2) `git checkout 248003eb && yarn install && yarn vite`, open file. broken.

**Comment by nerdCopter** [comment]
image

**Comment by nerdCopter** [comment]
existing logic: if no workspace exists, then get default workspace where default workspace is ctsnooze.

the appropriate files exist, so unsure why it breaks if prefs is not set.
image

**Comment by nerdCopter** [comment]
i found a workaround: checked out the commit prior commit (`ac8355d9`), loaded my personal workspace, checked out `master` (or `248003eb`), and the `workspaceGraphConfigs = upgradeWorkspaceFormat(item);` seemed to work.

but if user clears `storage` (browser-data) (with chrome extension), it breaks again.

**Comment by nerdCopter** [comment]
move this and it works without crash
```diff
diff --git a/src/main.js b/src/main.js
index e3a4ceeb..355b4d59 100644
--- a/src/main.js
+++ b/src/main.js
@@ -1064,6 +1064,9 @@ function BlackboxLogViewer() {
 
         graphLegend = new GraphLegend($(".log-graph-legend"), activeGraphConfig, onLegendVisbilityChange, onLegendSelectionChange, onLegendHighlightChange, zoomGraphConfig, expandGraphConfig, newGraphConfig);
 
+        workspaceSelection = new WorkspaceSelection($(".log-workspace-selection"), workspaceGraphConfigs, onSwitchWorkspace, onSaveWorkspace);
+        onSwitchWorkspace(workspaceGraphConfigs, workspaceSelection);
+
         // initial load of the configuration defaults if we have them
         prefs.get('workspaceGraphConfigs', function(item) {
             if (item) {
@@ -1082,9 +1085,6 @@ function BlackboxLogViewer() {
             }
         });
 
-        workspaceSelection = new WorkspaceSelection($(".log-workspace-selection"), workspaceGraphConfigs, onSwitchWorkspace, onSaveWorkspace);
-        onSwitchWorkspace(workspaceGraphConfigs, workspaceSelection);
-        
         workspaceMenu = new WorkspaceMenu($("#default_workspaces_menu"), onSwitchWorkspace);
 
         prefs.get('log-legend-hidden', function(item) {
```

but some other problems persist:
load a BBL without a workspace, graphs do not populate (no default workspace).  load a workspace after a BBL, graphs do not refresh and must reload the BBL to render.
still do not see Supfly and ctsnooze workspaces 
image

**Comment by nerdCopter** [comment]
this diff is a little better, but some same results as the hidden comment above.
```diff
diff --git a/src/main.js b/src/main.js
index e3a4ceeb..57d2c147 100644
--- a/src/main.js
+++ b/src/main.js
@@ -1064,6 +1064,8 @@ function BlackboxLogViewer() {
 
         graphLegend = new GraphLegend($(".log-graph-legend"), activeGraphConfig, onLegendVisbilityChange, onLegendSelectionChange, onLegendHighlightChange, zoomGraphConfig, expandGraphConfig, newGraphConfig);
 
+        workspaceMenu = new WorkspaceMenu($("#default_workspaces_menu"), onSwitchWorkspace);
+
         // initial load of the configuration defaults if we have them
         prefs.get('workspaceGraphConfigs', function(item) {
             if (item) {
@@ -1084,7 +1086,7 @@ function BlackboxLogViewer() {
 
         workspaceSelection = new WorkspaceSelection($(".log-workspace-selection"), workspaceGraphConfigs, onSwitchWorkspace, onSaveWorkspace);
         onSwitchWorkspace(workspaceGraphConfigs, workspaceSelection);
-        
+
         workspaceMenu = new WorkspaceMenu($("#default_workspaces_menu"), onSwitchWorkspace);
 
         prefs.get('log-legend-hidden', function(item) {
```

**Comment by demvlad** [comment]
Ok. Will check soon.

**Pull request merged**